### PR TITLE
Mopping Puddle Dilution and Wet Floor Sparkles

### PR DESCRIPTION
**Pull request by Willhelm53**
## About the PR 
***Note to maintainers:*** This PR builds on a lot of the files also changed in my last PR #6193. It may be easier to review this one once 6193 gets merged, but you'll know better than I will 🙂 
(My changelog notes are relative to that PR as well, so some squashing/combining may be needed)

### Wet Floors 💦⚠️
* Added back wet floors (My [[PR #6193]](https://github.com/space-wizards/space-station-14/pull/6193) temporarily broke/removed them)
    - Added a mopping lower limit of 5 units. Mops are now able to pick up all except the last 5 units of any puddle.  
    - The remaining 5 units can't be removed except by waiting for evaporation, leaving the floor nice and wet for the HoS to slip on.  ✊

### Sparkle Decal ✨
* Any time a puddle is mopped down to the 5u limit, add a sparkle decal to:  
    - Provide feedback to the janitor, signalling that they can move on to the next puddle  
    - Warn bystanders of the wet floor, I suppose 🙄  
* Miscellaneous notes:
    - Current implementation of this is that any puddle below 5u will now be seen as a sparkle, not just those left behind by mops.
* The sparkle is removed automatically when the puddle evaporates. This is because the effect is just a puddle sprite change using PuddleVisualizer. Thanks to @Peptide90 for the idea to just make it a sprite change, and to @mirrorcult for pointing me in the right direction to implement said sprite change.
    
### Diluting Contaminants 🩸➡️💧
* Based on the new mopping lower limit:
    * If a puddle of 5 units or smaller is clicked on with a mop, up to 10 units of solution from the mop are added to the puddle. This allows for nastier chemicals to be diluted and picked up, e.g.:
        * We start with a 5 unit puddle of fuel
        * It is clicked by a mop with water, becoming 5u fuel + 10u water
        * It is clicked by the mop again, which takes back 10 units of mixed solution.
        * The puddle is now 5 units (1.67 of fuel and 3.33 units of water)
* When a dry mop interacts with a mop bucket, it now becomes half-full instead of full. This makes it easier to go and dilute a puddle and pick it up without having to return to the bucket prematurely and dump mostly water.

**Screenshots**

image
image

**Changelog**

:cl:
- add: Added new sparkle sprite that is applied to puddles below a certain volume. Better look out for those wet floor signs!
- fix: Nerfed regular blue buckets to only hold 100u (down from 500u), bringing them more in line with their capacity in SS13.
- tweak: Mops can now pick up all but the last 5u of a puddle. The remaining 5u is how wet floors work. But hey, that's what evaporation is for!

### CODE REVIEW

File: Content.Client/Fluids/PuddleVisualizer.cs [hunk 1]
```diff
@@ -43,18 +43,19 @@ public override void InitializeEntity(EntityUid entity)
 
         public override void OnChangeData(AppearanceComponent component)
         {
-            base.OnChangeData(component);
+            base.OnChangeData(component); // runs OnChangeData from the base class (AppearanceVisualizer)
```
Comment (metalgearsloth) [inline Content.Client/Fluids/PuddleVisualizer.cs:46]: Unnecessary comment IMO.

File: Resources/Textures/Fluids/wet_floor_sparkles.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "tgstation commit 505e4414959d598c0d677a9bc734067a7bb513cd",
```
Comment (metalgearsloth) [inline Resources/Textures/Fluids/wet_floor_sparkles.rsi/meta.json:4]: link the repo.

File: Content.Server/Fluids/Components/MopComponent.cs [hunk 3]
```diff
@@ -72,10 +72,16 @@ public FixedPoint2 MaxVolume
...
+        ///     When using the mop on an empty floor tile, leave this much reagent as a new puddle.
         /// </summary>
         [DataField("residueAmount")]
-        public FixedPoint2 ResidueAmount { get; } = FixedPoint2.New(5);
+        public FixedPoint2 ResidueAmount { get; } = FixedPoint2.New(10); // Should be higher than MopLowerLimit
+
+        /// <summary>
+        ///     To leave behind a wet floor, the mop will be unable to take from puddles with a volume less than this amount.
+        /// </summary>
+        [DataField("MopLowerLimit")]
```
Comment (metalgearsloth) [inline Content.Server/Fluids/Components/MopComponent.cs:83]:
```suggestion
        [DataField("mopLowerLimit")]
```

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Willhelm53** [comment]
I pushed the requested changes and resolved conversations, should be all good now. Still new to Github so let me know if there are any other steps on my end. 🙂

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**